fastjet/internal/delaunay/point.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"math"
+	"strconv"
+)
+
+// Point in the X-Y Plane, that holds dynamic information
+type Point struct {
+	X       float64     // X coordinate
+	Y       float64     // Y coordinate
+	adjT    []*Triangle // holds all triangles adjacent to a point
+	outside bool        // true if point is on the outside of the triangulation
+	Ind     int         // user defined identifier
```

Comment:
If it needs to be there, then it needs to be there, but if it doesn't it shouldn't be.

From the way you describe its use, with no invariants, then a pointer value would do, but I see in `inPolygon` there clearly are invariants that are not supported by the use of pointer comparison; it seems to me from reading that that IDs must be unique within the set of points - this would work using pointer values, but it is not safe since the pointer values may change under foot (and pointer inequality is not completely defined). You should make this clear in the documentation.

Can you outline how you plan to use this in your problem so I can think about it more in context.